src/nft/components/bag/BagFooter.test.tsx

```diff
@@ -0,0 +1,415 @@
+    })
+    mocked(useTokenBalance).mockReturnValue(undefined)
+  })
+
+  it('pay', () => {
+    renderBagFooter()
+    const buyButton = getBuyButton()
+
+    expect(buyButton).toBeInTheDocument()
+    expect(buyButton.textContent).toBe('Pay')
+    expect(buyButton).not.toBeDisabled()
+  })
+
+  it('wallet not connected', () => {
+    mocked(useWeb3React).mockReturnValue({
+      chainId: 1,
+      account: undefined,
+    } as ReturnType<typeof useWeb3React>)
+
+    renderBagFooter()
+    const buyButton = getBuyButton()
+
+    expect(buyButton).toBeInTheDocument()
+    expect(buyButton.textContent).toBe('Connect wallet')
+    expect(buyButton).not.toBeDisabled()
+  })
+
+  it('connected to wrong network', () => {
+    mocked(useWeb3React).mockReturnValue({
+      chainId: 2,
+      account: '0x52270d8234b864dcAC9947f510CE9275A8a116Db',
+    } as ReturnType<typeof useWeb3React>)
+
+    renderBagFooter()
+    const buyButton = getBuyButton()
+
+    expect(buyButton.textContent).toBe('Switch networks')
+    expect(buyButton).not.toBeDisabled()
+  })
+
+  it('insufficient balance', () => {
+    mocked(useWalletBalance).mockReturnValue({
+      address: '',
+      balance: '0',
+      weiBalance: parseEther('0'),
+      provider: undefined,
+    })
+
+    renderBagFooter()
+    const buyButton = getBuyButton()
+    const buyButtonWarning = getBuyButtonWarning()
+
+    expect(buyButtonWarning.textContent).toBe('Insufficient funds')
+    expect(buyButton).toBeDisabled()
+  })
+
+  it('transaction error', () => {
+    mocked(useBag).mockReturnValue({
+      bagStatus: BagStatus.WARNING,
+      setBagStatus: () => undefined,
+      setBagExpanded: () => undefined,
+      isLocked: false,
+      itemsInBag: [],
+    }) as ReturnType<typeof useBag>
+
+    renderBagFooter()
+    const buyButtonWarning = getBuyButtonWarning()
+    const buyButton = getBuyButton()
+
+    expect(buyButtonWarning.textContent).toBe('Something went wrong. Please try again.')
+    expect(buyButton).toBeDisabled()
+  })
+
+  it('is in wallet confirmation for fetching route', () => {
+    mocked(useBag).mockReturnValue({
+      bagStatus: BagStatus.FETCHING_FINAL_ROUTE,
+      setBagStatus: () => undefined,
+      setBagExpanded: () => undefined,
+      isLocked: false,
+      itemsInBag: [],
+    }) as ReturnType<typeof useBag>
+
+    renderBagFooter()
+    const buyButton = getBuyButton()
+
+    expect(buyButton.textContent).toBe('Proceed in wallet')
+    expect(buyButton).toBeDisabled()
+  })
+
+  it('is in wallet confirmation for confirming in wallet', () => {
+    mocked(useBag).mockReturnValue({
+      bagStatus: BagStatus.FETCHING_FINAL_ROUTE,
+      setBagStatus: () => undefined,
+      setBagExpanded: () => undefined,
+      isLocked: false,
+      itemsInBag: [],
+    }) as ReturnType<typeof useBag>
+
+    renderBagFooter()
+    const buyButton = getBuyButton()
+
+    expect(buyButton.textContent).toBe('Proceed in wallet')
+    expect(buyButton).toBeDisabled()
+  })
```

Comment:
these two look identical

Reply:
mb - was supposed to be `CONFIRMING_IN_WALLET` for the bag status